### PR TITLE
Get all passing Metal tests in CI

### PR DESCRIPTION
**Pull request by duyle-py**
Get all passing Metal tests in CI
I found the bounty [here](https://docs.google.com/spreadsheets/d/1WKHbT-7KOgjEawq5h5Ic1qUWzpfAzuD_J06N1JwOCGs/edit#gid=0&range=A7) so want to fix it.

### CODE REVIEW

File: tinygrad/runtime/ops_metal.py [hunk 1]
```diff
@@ -44,7 +44,8 @@ def compile_metal(prg, use_xcode=bool(getenv("METAL_XCODE"))) -> bytes:
     # NOTE: if you run llvm-dis on "air" you can see the llvm bytecode
     air = subprocess.check_output(['xcrun', '-sdk', 'macosx', 'metal', '-x', 'metal', '-c', '-', '-o', '-'], input=prg.encode('utf-8'))
     return subprocess.check_output(['xcrun', '-sdk', 'macosx', 'metallib', '-', '-o', '-'], input=air)
-  options = Metal.MTLCompileOptions.alloc().init()
+  options = Metal.MTLCompileOptions.new()
+  options.setFastMathEnabled_(False)
```
Comment (geohot) [inline tinygrad/runtime/ops_metal.py:48]: How much impact does this have to speed?

Comment (duyle-py) [reply tinygrad/runtime/ops_metal.py:48]: Ok. Not sure. I will benchmark it.

Comment (duyle-py) [reply tinygrad/runtime/ops_metal.py:48]: I run `external_model_benchmark` on two codes. The results are not different. 

On duyle-py:master:
```
➜  tinygrad git:(0412dafc) ✗ python3 test/external/external_model_benchmark.py                
resnet50        tinygrad_metal_jitless      74.07 ms
resnet50        tinygrad_metal_jit          28.16 ms
resnet50        tinygrad_clang_jitless     945.40 ms
resnet50        tinygrad_clang_jit         877.01 ms
resnet50        torch_cpu                   71.40 ms
resnet50        torch_mps                   28.65 ms
resnet50        onnxruntime_cpu             41.62 ms
resnet50        outputs validated with rtol=2.0e-03, atol=2.0e-03
openpilot       tinygrad_metal_jitless     214.70 ms
openpilot       tinygrad_metal_jit           7.18 ms
openpilot       tinygrad_clang_jitless     275.78 ms
openpilot       tinygrad_clang_jit         156.68 ms
openpilot       onnx2torch              RuntimeError
openpilot       onnxruntime_cpu             46.40 ms
openpilot       outputs validated with rtol=2.0e-03, atol=2.0e-03
efficientnet    tinygrad_metal_jitless     270.41 ms
efficientnet    tinygrad_metal_jit          22.65 ms
efficientnet    tinygrad_clang_jitless     594.11 ms
efficientnet    tinygrad_clang_jit         330.55 ms
efficientnet    torch_cpu                  822.59 ms
efficientnet    torch_mps                   19.29 ms
efficientnet    onnxruntime_cpu             60.48 ms
efficientnet    outputs validated with rtol=2.0e-03, atol=2.0e-03
shufflenet      tinygrad_metal_jitless      57.19 ms
shufflenet      tinygrad_metal_jit           3.82 ms
shufflenet      tinygrad_clang_jitless      78.32 ms
shufflenet      tinygrad_clang_jit          27.71 ms
shufflenet      torch_cpu                   34.73 ms
shufflenet      torch_mps                   73.22 ms
shufflenet      onnxruntime_cpu              5.39 ms
shufflenet      outputs validated with rtol=2.0e-03, atol=2.0e-03
commavq         tinygrad_metal_jitless     207.90 ms
commavq         tinygrad_metal_jit          20.38 ms
commavq         tinygrad_clang_jitless     444.61 ms
commavq         tinygrad_clang_jit         252.42 ms
commavq         onnx2torch       NotImplementedError
commavq         onnxruntime_cpu             78.59 ms
commavq         outputs validated with rtol=2.0e-03, atol=2.0e-03
```
image

On tinygrad:master: 
```
➜  tinygrad git:(0c9b4ab8) ✗ python3 test/external/external_model_benchmark.py        
resnet50        tinygrad_metal_jitless      76.24 ms
resnet50        tinygrad_metal_jit          28.18 ms
resnet50        tinygrad_clang_jitless     931.83 ms
resnet50        tinygrad_clang_jit         869.03 ms
resnet50        torch_cpu                   74.77 ms
resnet50        torch_mps                   27.04 ms
resnet50        onnxruntime_cpu             39.73 ms
resnet50        outputs validated with rtol=2.0e-03, atol=2.0e-03
openpilot       tinygrad_metal_jitless     206.38 ms
openpilot       tinygrad_metal_jit           6.95 ms
openpilot       tinygrad_clang_jitless     262.44 ms
openpilot       tinygrad_clang_jit         155.86 ms
openpilot       onnx2torch              RuntimeError
openpilot       onnxruntime_cpu             46.52 ms
openpilot       outputs validated with rtol=2.0e-03, atol=2.0e-03
efficientnet    tinygrad_metal_jitless     273.18 ms
efficientnet    tinygrad_metal_jit          22.61 ms
efficientnet    tinygrad_clang_jitless     581.97 ms
efficientnet    tinygrad_clang_jit         328.71 ms
efficientnet    torch_cpu                  805.58 ms
efficientnet    torch_mps                   18.33 ms
efficientnet    onnxruntime_cpu             60.14 ms
efficientnet    outputs validated with rtol=2.0e-03, atol=2.0e-03
shufflenet      tinygrad_metal_jitless      57.67 ms
shufflenet      tinygrad_metal_jit           4.50 ms
shufflenet      tinygrad_clang_jitless      77.96 ms
shufflenet      tinygrad_clang_jit          27.68 ms
shufflenet      torch_cpu                   34.68 ms
shufflenet      torch_mps                   74.99 ms
shufflenet      onnxruntime_cpu              5.39 ms
shufflenet      outputs validated with rtol=2.0e-03, atol=2.0e-03
commavq         tinygrad_metal_jitless     209.29 ms
commavq         tinygrad_metal_jit          19.11 ms
commavq         tinygrad_clang_jitless     445.21 ms
commavq         tinygrad_clang_jit         250.85 ms
commavq         onnx2torch       NotImplementedError
commavq         onnxruntime_cpu             76.97 ms
commavq         outputs validated with rtol=2.0e-03, atol=2.0e-03
```
image

**Comment by duyle-py** [comment]
I enabled the metal tests in the new commit.

File: tinygrad/renderer/cstyle.py [hunk 2]
```diff
@@ -35,7 +35,8 @@ class CStyleLanguage(NamedTuple):
     UnaryOps.SIN: lambda x: f"sin({x})",
     UnaryOps.SQRT: lambda x: f"sqrt({x})",
     BinaryOps.ADD: lambda a,b: f"({a}+{b})", BinaryOps.SUB: lambda a,b: f"({a}-{b})",
-    BinaryOps.MUL: lambda a,b: f"({a}*{b})", BinaryOps.DIV: lambda a,b: f"({a}/{b})",
+    BinaryOps.MUL: lambda a,b: f"({a}*{b})" if "INFINITY" not in [a,b] else (f"(({a}>=0?1:-1)*({b}>=0?1:-1)*INFINITY)"),
```
Comment (geohot) [inline tinygrad/renderer/cstyle.py:38]: Why is this needed?

**Comment by geohot** [comment]
Can you try `test/models/test_train.py::TestTrain::test_vit` also?

File: tinygrad/renderer/cstyle.py (hunk 2 above)
Comment (duyle-py) [reply tinygrad/renderer/cstyle.py]: I thought that is a problem when multiplying values. But I'm wrong, diskcache uses the old program code.
So, using `options.setFastMathEnabled_(False)` is enough
I updated a new commit.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -215,9 +215,8 @@ jobs:
       run: pip install -e '.[metal,webgpu,testing]' --extra-index-url https://download.pytorch.org/whl/cpu
     - name: Test LLaMA compile speed
       run: PYTHONPATH="." METAL=1 python test/external/external_test_speed_llama.py
-    #- name: Run dtype test
-    #  run: DEBUG=4 METAL=1 python -m pytest -n=auto test/test_dtype.py
-    # dtype test has issues on test_half_to_int8
+    - name: Run dtype test
+      run: DEBUG=4 METAL=1 python -m pytest -n=auto test/test_dtype.py
```
Comment (chenyuxyz) [inline .github/workflows/test.yml:219]: also not all tests are included here, should consolidate the individual ones under `test/` to `python -m pytest -n=auto test/` like other backends

Comment (duyle-py) [reply .github/workflows/test.yml]: Ah. Understand. I will update it. Thank you!

**Comment by geohot** [comment]
Metal tests failing
